src/main/java/org/spoken_tutorial/health/elasticsearch/controllers/HomeController.java

```diff
@@ -0,0 +1,294 @@
+     * 
+     * try { docContent = ContentsfromFile.extractContentForElastic(parser, path);
+     * tut.setDocumentContent(docContent);
+     * 
+     * tutorialRepo.save(tut); } catch (Exception e) {
+     * 
+     * }
+     * 
+     * return 1;
+     * 
+     * }
+     * 
+     */
+
+    @PostMapping("/addDocument/{documentId}/{documentType}/{documentPath}/{documenttUrl}/{rank}/{view_url}/{language}")
+    public Map<Long, String> addDocument(@PathVariable String documentId, @PathVariable String documentType,
+            @PathVariable String documentPath, @PathVariable String documenttUrl, @PathVariable int rank,
+            @PathVariable String view_url, @PathVariable String language, @RequestParam String category,
+            @RequestParam String topic, @RequestParam String outlinePath) {
+
+        Map<Long, String> resultMap = new HashMap<>();
+        long queueId = queuemntService.getNewId();
+
+        if (doesFileExist(documentPath)) {
+
+            QueueManagement queuemnt = new QueueManagement();
+            queuemnt.setQueueId(queueId);
+            queuemnt.setRequestId(documentType + documentId);
+            queuemnt.setRequestTime(getCurrentTime());
+            queuemnt.setRequestType("addDocument");
+            queuemnt.setDocumentId(documentId);
+            queuemnt.setDocumentType(documentType);
+            queuemnt.setDocumentId(documentPath);
+            queuemnt.setDocumentUrl(documenttUrl);
+            queuemnt.setRank(rank);
+            queuemnt.setView_url(view_url);
+            queuemnt.setLanguage(language);
+            queuemnt.setMessage("success");
+
+            Optional category1 = Optional.ofNullable(category);
+            if (category1.isPresent()) {
+                queuemnt.setCategory((String) category1.get());
+            }
+
+            Optional topic1 = Optional.ofNullable(topic);
+            if (topic1.isPresent()) {
+                queuemnt.setTopic((String) topic1.get());
+            }
+
+            Optional outlinePath1 = Optional.ofNullable(outlinePath);
+            if (outlinePath1.isPresent()) {
+                if (doesFileExist((String) outlinePath1.get())) {
+                    queuemnt.setOutlinePth((String) outlinePath1.get());
+                }
+
+            }
+
+            queRepo.save(queuemnt);
+            resultMap.put(queueId, "success");
+        }
+
+        else {
+            resultMap.put(queueId, "failed");
+        }
+
+        return resultMap;
+
+    }
+
+    @PutMapping("/updateDocument/{documentId}/{documentType}/{documentPath}")
+    public Map<Long, String> updateDocument(@PathVariable String documentId, @PathVariable String documentType,
+            @PathVariable String documentPath, @RequestParam String category, @RequestParam String topic,
+            @RequestParam String outlinePath) {
+
+        QueueManagement queuemnt = queRepo.findByDocumentTypeAndDocumentId(documentType, documentId);
```

Comment:
this should not update the database. it will just add an updateDocument request to db.

code will be similar to addDocument except for the requestType